app/utils/stageUtils.ts

```
@@ -64,6 +64,28 @@ export default class StageUtils {
      */
     static formatLocalTimestamp = TimeUtils.formatLocalTimestamp.bind(TimeUtils);
 
+    /**
+     * Formatting resource id and display name.
+     *
+     * @param data
+     * @returns string
+     */
+    static formatDisplayName(data: Partial<{ id: string; displayName: string }>): string {
+        if (data.id === undefined) {
+            throw new Error('id is undefined!');
+        }
```

We can validate this condition with TypeScript if we do:

```ts
data: { id: string; displayName?: string }
```

in the parameter :smile:

---

no, in some cases id is `undefined` don't know why but it is (checked by running system tests and throwing error when is undefined. and it's outdated

---

Well, then IMO we should fix these places rather than ignore them :smile: If we expect to show deployment names and the function fails, that's a silent error that, to me, is worse than an explicit failure :smile:

---

Also, if you don't exactly apply my suggestion, could you not resolve the conversation to let me answer back :slightly_smiling_face: ? This is the usual flow we follow during code review

---

Added log.error in this case

---

Hmm I don't know if that's enough :smile: I think we should fix these places instead of having a broken UI (the UI won't display the ID or name in these cases)